compiler/noirc_evaluator/src/brillig/brillig_ir/procedures/vector_pop_front.rs

```diff
@@ -76,49 +76,50 @@ pub(super) fn compile_vector_pop_front_procedure<F: AcirField + DebugToString>(
         1_usize,
     );
 
-    brillig_context.codegen_branch(is_rc_one, |brillig_context, is_rc_one| {
-        if is_rc_one {
-            // We reuse the source vector, moving the metadata to the right (decreasing capacity)
-            brillig_context.memory_op_instruction(
-                source_vector.pointer,
-                item_pop_count_arg,
-                target_vector.pointer,
-                BrilligBinaryOp::Add,
-            );
-            brillig_context.memory_op_instruction(
-                source_capacity.address,
-                item_pop_count_arg,
-                source_capacity.address,
-                BrilligBinaryOp::Sub,
-            );
-            brillig_context.codegen_initialize_vector_metadata(
-                target_vector,
-                target_size,
-                Some(source_capacity),
-            );
-        } else {
-            brillig_context.codegen_initialize_vector(target_vector, target_size, None);
-
-            let target_vector_items_pointer =
-                brillig_context.codegen_make_vector_items_pointer(target_vector);
-
-            let source_copy_pointer = brillig_context.allocate_register();
-            brillig_context.memory_op_instruction(
-                source_items_pointer.address,
-                item_pop_count_arg,
-                source_copy_pointer,
-                BrilligBinaryOp::Add,
-            );
-            // Now we copy the source vector starting at index removed_items.len() into the target vector
-            brillig_context.codegen_mem_copy(
-                source_copy_pointer,
-                target_vector_items_pointer,
-                target_size,
-            );
-
-            brillig_context.deallocate_register(source_copy_pointer);
-            brillig_context.deallocate_register(target_vector_items_pointer);
-        }
+    brillig_context.codegen_branch(is_rc_one, |brillig_context, _is_rc_one| {
```

Comment:
Do we even need the branch if we are not checking `is_rc_one`

Reply:
Removed